src/index.ts

```diff
@@ -386,9 +386,54 @@ export async function install(
     chunkFileNames: 'common/[name]-[hash].js',
   };
   if (Object.keys(depObject).length > 0) {
-    const packageBundle = await rollup.rollup(inputOptions);
-    await packageBundle.write(outputOptions);
+    // Main Snowpack Install 🚀
+    try {
+      const packageBundle = await rollup.rollup(inputOptions);
+      await packageBundle.write(outputOptions);
+    } catch (err) {
+      // Handle Snowpack Install error
+      const invalidFile = err.loc.file as string;
+
+      // handle invalid files
+      if (invalidFile) {
+        const ext = invalidFile.substr(invalidFile.lastIndexOf('.')).toLowerCase();
+        const relativeName = invalidFile.replace(cwd, '');
+        const suggest = (suggestion: string) =>
+          logError(
+            `${chalk.bold(
+              'snowpack',
+            )} encountered an error importing ${relativeName}. Try ${suggestion}`,
+          );
+
+        // based on the file’s extension, suggest a course of action (or fall through to generic error below):
+        switch (ext) {
+          case '.css': {
+            suggest(
+              'installing rollup-plugin-postcss and adding it to Snowpack (https://www.snowpack.dev/#custom-rollup-plugins)',
+            );
+            return;
+          }
+          case '.jsx': {
+            suggest(
+              'installing @babel/preset-react and enabling Babel (https://www.snowpack.dev/#babel)',
+            );
+            return;
+          }
+          case '.svelte': {
+            suggest(
+              'installing rollup-plugin-svelte and adding it to Snowpack (https://www.snowpack.dev/#custom-rollup-plugins)',
+            );
+            return;
+          }
+        }
+      }
```

Comment:
The 3 suggestions from the issue are here:

| Extension | Suggestion |
|----------|-------------|
| `.css`     | `rollup-plugin-postcss` (I confirmed this works, and it does) |
| `.jsx`      | Babel + `@babel/preset-react` |
| `.svelte` | `rollup-plugin-svelte` (tested in another test) |

I know there are probably more, but I can’t think of any. Should there be any others to start?

Reply:
Maybe `.vue`, `.less` and `.s[a|c]ss`?

Also, maybe `.tsx`?

Reply:
Oh good call! Adding those.

Reply:
Looks like `postcss` plugin can handle `.css .pcss .sss .styl .scss .sass .less`

Reply:
+1 for all of the above except for tsx, that starts to go down a 'set up a tsconfig.json' and 'install  TypeScript' rabbit hole that I'm worried might end up confusing a user more than just erroring. I might be wrong on that, but either way I'd prefer to have an example of a user hitting that (reported to us as an issue/discussion) before tackling.

Reply:
You know, I use the PostCSS CLI all the time, and the Sass CLI all the time, but I’ve never compiled Sass or Less with `rollup-plugin-postcss`. In this specific scenario I don’t have much of an opinion.

If `rollup-plugin-postcss` can handle all of the above, I think that makes sense to make one suggestion since it’s so well-used and maintained. At a cursory glance it seems to have more users than the individual Sass and Less Rollup plugins.

Reply:
>+1 for all of the above except for tsx, that starts to go down a 'set up a tsconfig.json' and 'install TypeScript' rabbit hole

Ah totally see where you’re coming from. So what should the `.ts` warning say then? It can say anything.

Reply:
Actually, thinking about it more I'd be worried about that same SASS/LESS problem (postcss is close to SASS but not a 1:1 perfectly complete replacement irrc). 

I don't think many packages ship SASS/LESS without CSS, and I'd be curious what the use-case to do that would be, so maybe skip for now and see if anyone reports it as an issue?

Reply:
I'm good with whatever you guys decide. I was just throwing out some common extensions. I think this is a nice DX win either way. We can add more if we get enough feedback around missing extension support.

Reply:
>I don't think many packages ship SASS/LESS without CSS

You have a point there—these would only be encountered by a JS file importing them, which would be weird. I think it probably _should_ give an error, but the suggested course of action is unclear. Maybe the error should be more of a generic `Can’t import .sass from JS` error that makes the problem clear, while not suggesting what may be a bad course of action?

Reply:
100% okay with whatever you decide is best here

Reply:
Wanted to think on it. I think in regard to Sass/Less and TS, I’d rather not offer suggestions for now, and wait for any issues to be raised. Because if someone is trying to import those into an ESM environment, likely it’s against the developer’s knowledge, and likely it’s some npm package that’s doing unexpected things.

I’d rather offer suggestions informed by those scenarios rather than try and guess why someone would try and build an ESM app with Sass. So I think just the suggestions in this file are good for now